### ISSUE TITLE
Invalid schema while listing runtime configurations via metadata cli

### ISSUE THREAD

**Issue by lresende** [elyra-ai/elyra#1355]
**Describe the issue**
Invalid schema while listing runtime configurations via metadata CLI

**To Reproduce**
1. `elyra-metadata list runtimes`
```
Available metadata instances for runtimes (includes invalid):

Schema    Instance               Resource                                                                      
------    --------               --------                                                                      
airflow   apache_airflow_v11012  /.../Library/Jupyter/metadata/runtimes/apache_airflow_v11012.json  **INVALID** (SchemaNotFoundError)
kfp       kfp-shared             /.../Library/Jupyter/metadata/runtimes/kfp-shared.json             
kfp       kfp-shared-argo        /.../Library/Jupyter/metadata/runtimes/kfp-shared-argo.json        
kfp       kfp-shared-tekton      /.../Library/Jupyter/metadata/runtimes/kfp-shared-tekton.json
```

2. `more /.../Library/Jupyter/metadata/runtimes/apache_airflow_v11012.json`

```
{
  "display_name": "Apache Airflow v1.10.12",
  "metadata": {
    "api_endpoint": "http://teacloth1.fyre.ibm.com:32706/",
    "github_api_endpoint": "xxx",
    "github_repo": "xxx",
    "github_repo_token": "xxx",
    "github_branch": "main",
    "cos_endpoint": "xxx",
    "cos_username": "xxx",
    "cos_password": "xxx",
    "cos_bucket": "elyra-pipeline",
    "tags": []
  },
  "schema_name": "airflow"
}

```
And, even though this is marked as invalid, it is available via REST APIs.

**Comment by kevin-bates** [comment]
IIRC, we tolerate a missing schema but not an invalid instance wrt exposure via REST.

Any idea why the airflow schema is not present?

**Comment by lresende** [comment]
> Any idea why the airflow schema is not present?
That is exactly the question, to me, it is there, but maybe it's getting confused with two schemas now?
Having said that, I did not investigate the issue yet.

**Comment by kevin-bates** [comment]
Ok - thanks, I'll take a look.  I wanted to make sure this wasn't an older release or something funky going on.

**Comment by kevin-bates** [comment]
Hmm.  I'm getting what appears to be a valid `ValidationError`...
```
$ elyra-metadata list runtimes
[E 2021-03-02 09:34:31,814.814] Validation failed for instance 'airflow_1' using the airflow schema with error: 'xxx' is not a 'uri'.
Available metadata instances for runtimes (includes invalid):

Schema    Instance      Resource                                                           
------    --------      --------                                                           
airflow   airflow_1     /Users/kbates/Library/Jupyter/metadata/runtimes/airflow_1.json     **INVALID** (ValidationError)
kfp       baz           /Users/kbates/Library/Jupyter/metadata/runtimes/baz.json           
kfp       foofoo        /Users/kbates/Library/Jupyter/metadata/runtimes/foofoo.json        
kfp       kf_1          /Users/kbates/Library/Jupyter/metadata/runtimes/kf_1.json          
kfp       kf_eg         /Users/kbates/Library/Jupyter/metadata/runtimes/kf_eg.json         
kfp       kfp-dex-auth  /Users/kbates/Library/Jupyter/metadata/runtimes/kfp-dex-auth.json  


$ cat airflow_1.json 
{
  "display_name": "Apache Airflow v1.10.12",
  "metadata": {
    "api_endpoint": "http://teacloth1.fyre.ibm.com:32706/",
    "github_api_endpoint": "xxx",
    "github_repo": "xxx",
    "github_repo_token": "xxx",
    "github_branch": "main",
    "cos_endpoint": "xxx",
    "cos_username": "xxx",
    "cos_password": "xxx",
    "cos_bucket": "elyra-pipeline",
    "tags": []
  },
  "schema_name": "airflow"
}
```

Once I fix the validation errors (there are 3, two URIs are invalid (`github_api_endpoint` and `cos_endpont`) and `cos_password` is too short) then the command completes:
```
$ elyra-metadata list runtimes
Available metadata instances for runtimes (includes invalid):

Schema    Instance      Resource                                                           
------    --------      --------                                                           
airflow   airflow_1     /Users/kbates/Library/Jupyter/metadata/runtimes/airflow_1.json     
kfp       baz           /Users/kbates/Library/Jupyter/metadata/runtimes/baz.json           
kfp       foofoo        /Users/kbates/Library/Jupyter/metadata/runtimes/foofoo.json        
kfp       kf_1          /Users/kbates/Library/Jupyter/metadata/runtimes/kf_1.json          
kfp       kf_eg         /Users/kbates/Library/Jupyter/metadata/runtimes/kf_eg.json         
kfp       kfp-dex-auth  /Users/kbates/Library/Jupyter/metadata/runtimes/kfp-dex-auth.json  
```
This was after pulling `master` and `make clean install`.

**Comment by kevin-bates** [comment]
I just reproduced this using the released elyra - 2.0.1.  
```
$ elyra-metadata list runtimes
Available metadata instances for runtimes (includes invalid):

Schema    Instance      Resource                                                           
------    --------      --------                                                           
airflow   airflow_1     /Users/kbates/Library/Jupyter/metadata/runtimes/airflow_1.json     **INVALID** (SchemaNotFoundError)
kfp       baz           /Users/kbates/Library/Jupyter/metadata/runtimes/baz.json           
kfp       foofoo        /Users/kbates/Library/Jupyter/metadata/runtimes/foofoo.json        
kfp       kf_1          /Users/kbates/Library/Jupyter/metadata/runtimes/kf_1.json          
kfp       kf_eg         /Users/kbates/Library/Jupyter/metadata/runtimes/kf_eg.json         
kfp       kfp-dex-auth  /Users/kbates/Library/Jupyter/metadata/runtimes/kfp-dex-auth.json  
```
The airflow schema is not present...
```
$ ls /opt/anaconda3/envs/elyra-dev/lib/python3.6/site-packages/elyra/metadata/schemas
code-snippet.json   kfp.json            metadata-test.json  metadata-test2.json runtime-image.json
```

Airflow was not released in 2.0.1, so I believe this is still behaving as expected.  The [inclusion of the schemas in the MANIFEST.in file](https://github.com/elyra-ai/elyra/blob/master/MANIFEST.in#L36) looks correct.

**Comment by lresende** [comment]
Thanks for helping investigate this @kevin-bates, for some reason, I cannot reproduce this anymore even though I didn't change anything on the actual metadata file. For now, I will close this, but we can revive it if we start seeing this again,
